cirq/google/devices/serializable_device_test.py

```diff
@@ -50,6 +52,14 @@
 )
 
 
+@pytest.mark.parametrize('cycle,func', [(False, str), (True, repr)])
+def test_repr_pretty(cycle, func):
+    device = cg.Sycamore
+    printer = mock.Mock()
+    device._repr_pretty_(printer, cycle)
```

Comment:
Doesn't this test basically just call the function and assert that it was called?

I was thinking we should test that the grid gets printed out?

Reply:
Well, we call one function and assert that *another* function was called. This is called interaction testing. It is generally considered a bit harder to maintain (e.g. a refactoring that yields the same result using a different interaction will need to modify the test), so +1 to Doug's suggestion.

Reply:
TBH I mostly added this test so the coverage checks wouldn't complain at me :-)

But I will add a separate test checking that `SerializableDevice.__str__` outputs a nice grid.

Reply:
+1 on not relying on interaction testing / mocks but rather actual behavior/output.